PyInstaller/building/api.py

```diff
@@ -181,6 +184,11 @@ def __init__(self, toc, name=None, cdict=None, exclude_binaries=0,
         self.exclude_binaries = exclude_binaries
         self.strip_binaries = strip_binaries
         self.upx_binaries = upx_binaries
+        if is_darwin:
+            from ..config import CONF
+            self.codesign_identity = codesign_identity if codesign_identity is not None else CONF['codesign_identity']
```

Comment:
I'm not exactly sure how else to pass command-line arguments forward as "defaults" to the various classes (EXE, PKG, osx.BUNDLE, etc).  Can you illuminate me on how this would be done?

I don't want users in the .spec files to have to manually enter:
```
EXE(...,codesign_identity="Bla bla")
#...
BUNDLE(...,codesign_identity="Bla bla bla")
# ..
```

When they could just invoke pyinstaller with `pyinstaller --osx-codesign-identity XXX`.

I'll go back and read the sources more and try and figure it out.  But any pointers on how to have "global defaults" that come from the command-line here would be really helpful.

Let me know!

Reply:
So we have different cases:

* for `PKG` (which is not exposed in the spec-file): the function argument should be used (passed by `EXE`, as already done [here](https://github.com/pyinstaller/pyinstaller/blob/15ad5b7d805d0905d0ce8cc5b01d19e992796d85/PyInstaller/building/api.py#L447)

* for `EXE` and `BUNDLE` we first need to decide 1) whether we should require the identity to be specified on the command line for actually signing or 2) whether i should be possible to specify the ID in the spec-file. **What do you think**?

  * **If 1**:   Both `EXE` and `BUNDLE` should **not** accept a command line argument, but take the value from `CONF` only`. This would avoid one is specifying two conflicting values.

  * **if 2**: The ID could be put into the .spec-file by adding the appropriate line to the templates `PyInstaller/building/templates.py` and `PyInstaller/building/makespec.py`.